samples/tutorials/mnist/02_Local_Development_with_Docker_Image_Components.ipynb

```diff
@@ -0,0 +1,548 @@
+   "source": [
+    "## Create client\n",
+    "\n",
+    "**If submit outside the kubeflow cluster, need the following**\n",
+    "- `host`: the host name to use to talk to Kubeflow Pipelines, i.e., \"https://`<your-deployment>`.endpoints.`<your-project>`.cloud.goog/pipeline\"\n",
+    "- `client_id`: The client ID used by Identity-Aware Proxy\n",
+    "- `other_client_id`: The client ID used to obtain the auth codes and refresh tokens.\n",
+    "- `other_client_secret`: The client secret used to obtain the auth codes and refresh tokens.\n",
+    "- For getting `other_client_id` and `other_client_secret`, you'll need to create OAuth client ID credentials of type `Other` according to the tutorial [here](\n",
+    "https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_desktop_app)\n",
+    "\n",
+    "```python\n",
+    "client = kfp.Client(host, client_id, other_client_id, other_client_secret)\n",
+    "```\n",
+    "\n",
+    "**If you run and submit within the kubeflow cluster**, the following is enough\n",
+    "```python\n",
+    "client = kfp.Client()\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create kfp client\n",
+    "in_cluster = True\n",
+    "try:\n",
+    "  k8s.config.load_incluster_config()\n",
+    "except:\n",
+    "  in_cluster = False\n",
+    "  pass\n",
+    "\n",
+    "if in_cluster:\n",
+    "    client = kfp.Client()\n",
+    "else:\n",
+    "    client = kfp.Client(host=HOST, \n",
+    "                        client_id=CLIENT_ID,\n",
+    "                        other_client_id=OTHER_CLIENT_ID, \n",
+    "                        other_client_secret=OTHER_CLIENT_SECRET)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrap an existing Docker container image using `ContainerOp`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a Docker container\n",
+    "Create your own container image that includes your program. \n",
+    "- If your component creates some outputs to be fed as inputs to the downstream components, each separate output must be written as a string to a separate local text file inside the container image. \n",
+    "- For example, if a trainer component needs to output the trained model path, it can write the path to a local file `/output.txt`. \n",
+    "- The string written to an output file cannot be too big. If it is too big (>> 100 kB), it is recommended to save the output to an external persistent storage and pass the storage path to the next component."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell creates a file `app.py` that contains a Python script. The script takes a GCS bucket name as an input argument, gets the lists of blobs in that bucket, prints the list of blobs and also writes them to an output file."
```

Comment:
prints the list of blobs and also writes them to an output file.

->

prints the list of blobs, and writes the blobs to an output file.

Reply:
Nit: consider defining blob, if you think the audience may not know it off-hand

Reply:
Hey, this part is actually copied from some of the old notebooks, which didn't describe what is actually included in the script. I have modified, including all other notebooks.

->

The following cell creates a file `app.py` that contains a Python script. The script downloads MNIST dataset, trains a Neural Network based classification model, writes the log and exports the trained model to Google Cloud Storage.